pkg/function/wait_for_snapshot_completion.go

```diff
@@ -65,22 +65,16 @@ func waitForSnapshotsCompletion(ctx context.Context, snapshotinfo string, profil
 	if err != nil {
 		return errors.Wrapf(err, "Could not decode JSON data")
 	}
+
 	for _, pvcInfo := range PVCData {
-		config := make(map[string]string)
 		if err = ValidateLocationForBlockstorage(profile, pvcInfo.Type); err != nil {
 			return errors.Wrap(err, "Profile validation failed")
 		}
-		switch pvcInfo.Type {
-		case blockstorage.TypeEBS:
+		config := getConfig(profile, pvcInfo.Type)
```

Comment:
`getConfig` is not handling unsupported storage provider. Will something else handle `default` case?

Reply:
Oh I assumed `ValidateLocationForBlockstorage ` func takes care of that. Let me update the PR.

Reply:
That also doesn't. It was called `ValidateProfile` before I renamed it. Do you think we should handle the check there?

Reply:
I think so, since it will fail at validation stage, I just added a commit to this PR for the change.